starfish/recipe/runnable.py

```diff
@@ -0,0 +1,254 @@
+import inspect
+import warnings
+from typing import (
+    Any,
+    Callable,
+    Mapping,
+    MutableMapping,
+    MutableSequence,
+    Sequence,
+    Set,
+    Type,
+)
+
+from starfish.pipeline.algorithmbase import AlgorithmBase
+from starfish.pipeline.pipelinecomponent import PipelineComponent, PipelineComponentType
+from .errors import (
+    ConstructorError,
+    ConstructorExtraParameterWarning,
+    ExecutionError,
+    RunInsufficientParametersError,
+    TypeInferenceError,
+)
+from .filesystem import FileProvider, TypedFileProvider
+
+
+class Runnable:
+    """Runnable represents a single invocation of a pipeline component, with a specific algorithm
+    implementation.  For arguments to the algorithm's constructor and run method, it can accept
+    :py:class:`starfish.recipe.filesystem.FileProvider` objects, which represent a file path or url.
```

Comment:
doc tip! Adding ~ before the classpath renders the link as just `FileProvider` so ```:py:class:`~starfish.recipe.filesystem.FileProvider```

Reply:
Done.